### ISSUE TITLE
Reconcile difference in overriding behavior between Guice and JSR-330's @Inject

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pombreda/google-guice#512]
```
Guice's old behavior: Always inject a method annotated with @Inject regardless 
of overriding. In fact, if an overridden method is annotated with @Inject, we 
inject the method twice.

JSR-330's behavior: Only ever inject a method once. If an overriding method 
doesn't have @Inject, don't inject. This enables users to disable injection.

--

We've currently adopted the JSR-330 behavior wholesale. We print a warning when 
the user overrides using Guice's @Inject.

I prefer to not break existing apps, and I prefer to avoid warnings. Most users 
ignore them. 

Here's how I think it should work:

  - Support old behavior for Guice's @Inject
  - Support new behavior for JSR-330 @Inject

In other words, if Guice's @Inject is present on any method in the hierarchy, 
we always inject.

```

Original issue reported on code.google.com by `crazybob...@gmail.com` on 1 Jul 2010 at 1:40
